expression/builtin_json_vec.go

```diff
@@ -967,11 +967,83 @@ func (b *builtinJSONContainsPathSig) vecEvalInt(input *chunk.Chunk, result *chun
 }
 
 func (b *builtinJSONArrayAppendSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinJSONArrayAppendSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	m := (len(b.args) - 1) / 2
+
+	jsonBufs, err := b.bufAllocator.get(types.ETJson, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(jsonBufs)
+	if err := b.args[0].VecEvalJSON(b.ctx, input, jsonBufs); err != nil {
+		return err
+	}
+
+	pathBufs := make([]*chunk.Column, 0, m)
+	valBufs := make([]*chunk.Column, 0, m)
+	defer func() {
+		for _, buf := range pathBufs {
+			b.bufAllocator.put(buf)
+		}
+		for _, buf := range valBufs {
+			b.bufAllocator.put(buf)
+		}
+	}()
+	for i := 1; i < len(b.args)-1; i += 2 {
+		pathBuf, err := b.bufAllocator.get(types.ETString, n)
+		if err != nil {
+			return err
+		}
+		pathBufs = append(pathBufs, pathBuf)
+		if err := b.args[i].VecEvalString(b.ctx, input, pathBuf); err != nil {
+			return err
+		}
+		valBuf, err := b.bufAllocator.get(types.ETJson, n)
+		if err != nil {
+			return err
+		}
+		if err := b.args[i+1].VecEvalJSON(b.ctx, input, valBuf); err != nil {
+			return err
+		}
+		valBufs = append(valBufs, valBuf)
+	}
+
+	result.ReserveJSON(n)
+	for i := 0; i < n; i++ {
+		if jsonBufs.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		res := jsonBufs.GetJSON(i)
+		isNull := false
+		for j := 0; j < m; j++ {
+			s, sNull := "", pathBufs[j].IsNull(i)
+			if !sNull {
+				s = pathBufs[j].GetString(i)
+			}
+			v, vNull := json.BinaryJSON{}, valBufs[j].IsNull(i)
+			if !vNull {
+				v = valBufs[j].GetJSON(i)
+			}
+			res, isNull, err = b.appendJSONArray(res, s, sNull, v, vNull)
```

Comment:
If sNull is true, set isNull is true and break the loop.
If vNull is true, set `v = json.CreateBinary(nil)`.
So we can only transfer 3 parameters for function `appendJSONArray`

Reply:
Fixed. PTAL